synapse/storage/data_stores/main/push_rule.py

```diff
@@ -702,12 +727,20 @@ def set_push_rule_actions_txn(txn, stream_id, event_stream_ordering):
                     update_stream=False,
                 )
             else:
-                self.db.simple_update_one_txn(
-                    txn,
-                    "push_rules",
-                    {"user_name": user_id, "rule_id": rule_id},
-                    {"actions": actions_json},
-                )
+                try:
+                    self.db.simple_update_one_txn(
+                        txn,
+                        "push_rules",
+                        {"user_name": user_id, "rule_id": rule_id},
+                        {"actions": actions_json},
+                    )
+                except SynapseError as serr:
+                    if serr.code == 404:
+                        raise SynapseError(
+                            404, "Push rule does not exist", Codes.NOT_FOUND
+                        )
+                    else:
+                        raise serr
```

Comment:
we want to allow the exception to propagate with its original stacktrace, rather than raising it as a new exception.

```suggestion
                        raise
```

Reply:
Thanks for the insight :)